### PR TITLE
Export students' scores data, etc.

### PR DESCRIPTION
**Pull request by mike-allaway**
Primarily includes a feature for downloading students' scores as an Excel spreadsheet, as well as the page that allows admins/teachers to choose which students to include by filtering on class.

Also includes other minor enhancements/fixes

### CODE REVIEW

File: django/exercises/views.py [hunk 1]
```diff
@@ -48,6 +51,44 @@ def get_context_data(self, **kwargs):
...
+    template_name = 'exercises/exportdata-studentscores-options.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # Return all classes if user is an admin
+        if self.request.user.role.name == 'admin':
+            context['classes'] = models.SchoolClass.objects.all()
+        elif self.request.user.role.name == 'teacher':
+            context['classes'] = models.SchoolClass.objects.filter(user__id=self.request.user.id)
```
Comment (garciaay) [inline django/exercises/views.py:70]: Is 'user__id' correct (double underscore)?

File: django/exercises/exportdata/studentscores/exportstudentscores.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    # Delete all existing files in the data folder
+    data_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'outputdata')
+    files = glob.glob(data_path + '/*')
+    for f in files:
+        os.remove(f)
+    # Establish new file name
+    file_name = f'lfa_df_export_studentscores_{time.strftime("%Y-%m-%d_%H-%M")}.xlsx'
+    file_path = os.path.join(data_path, file_name)
+    # Create xlsxwriter workbook
+    workbook = xlsxwriter.Workbook(file_path)
```
Comment (garciaay) [inline django/exercises/exportdata/studentscores/exportstudentscores.py:66]: Would it be helpful to wrap this in a try - catch?

**Review by garciaay — APPROVED** [review]
All looks sensible to me. I have put a couple of comments in, but happy to approve as is.

File: django/exercises/views.py (hunk 1 above)
Comment (mike-allaway) [reply django/exercises/views.py:70]: yeah, so double underscores are used to access fields of related data. so in this case user is a foreign key and I want to get its Id so it's user__id ... e.g. let's say that user has a role and I want to get the name of that role you'd use double underscore again, so user__role__name :)

File: django/exercises/exportdata/studentscores/exportstudentscores.py (hunk 2 above)
Comment (mike-allaway) [reply django/exercises/exportdata/studentscores/exportstudentscores.py:66]: The documentation for xslxwriter doesn't include try/except when creating a workbook. I don't mind if it fails here as it'd need to stop running at this point anyway as it couldn't carry on without creating the workbook and in the unlikely event it happens the user would just see a standard error page, which is handled by Django

**Pull request merged**